src/SAWScript/Builtins.hs

```diff
@@ -531,6 +546,76 @@ goal_eval unints =
      prop' <- io (evalProp sc unintSet (goalProp goal))
      return (prop', EvalEvidence unintSet)
 
+extract_uninterp ::
+  [String] {- ^ uninterpred identifiers -} ->
+  [String] {- ^ opaque identifiers -} ->
+  TypedTerm ->
+  TopLevel (TypedTerm, [(String,[(TypedTerm,TypedTerm)])])
+extract_uninterp unints opaques tt =
+  do sc <- getSharedContext
+     idxs <- mapM (resolveName sc) unints
+     let unintSet = Set.fromList idxs
+     mmap <- io (scGetModuleMap sc)
+
+     opaqueSet <- Set.fromList <$> mapM (resolveName sc) opaques
+
+     boundECRef <- io (newIORef Set.empty)
+     let ?recordEC = \ec -> modifyIORef boundECRef (Set.insert ec)
+     (tm, repls) <- io (TM.extractUninterp sc mmap mempty mempty unintSet opaqueSet (ttTerm tt))
+     boundECSet <- io (readIORef boundECRef)
+     let tt' = tt{ ttTerm = tm }
+
+     let f = traverse $ \(ec,vs) ->
+               do ectm <- scExtCns sc ec
+                  vs'  <- filterCryTerms sc vs
```

Comment:
This isn't strictly required, I think.  We have a couple of options here.

1) Do what we have here.  Under the assumption that the original term comes from Cryptol, it will omit only and exactly the type-level arguments (numeric or value types), leaving the value arguments in the tuple, which can then be projected in Cryptol splices via standard Cryptol projection syntax.

2) Don't filter, but still produce a SAWCore tuple. The resulting tuple will sometimes (often) not have a Cryptol type, and thus will be hard to use from a `{{ ... }}` splice.

3) Don't filter, but return a `[Term]` instead of building a tuple.  This then requires SAWCore manipulation, and is a bit harder to use in Cryptol splices, but not as hard as option 2.

Reply:
As to bizarre behavior, what would mainly happen is that we might omit some argument that the user actually wanted.

Reply:
I'm inclined to leave it as-is for the moment, as option 1 is probably the most common use case. Most other SAWCore terms that we actually deal with can be given Cryptol types, too (e.g., anything from What4). If we start dealing with use cases where 2 or 3 would be more convenient, we can adapt. (And it probably means that the user-visible features should stay experimental for now.)